cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/statementHandler/StatementHandlerTest.kt

```diff
@@ -123,4 +119,19 @@ class StatementHandlerTest {
         assertNotNull(message, "Assert statement should have a message")
         assertEquals("Test message", message.value, "The assert message is incorrect")
     }
+
+    @Test
+    fun testTypeHints() {
+        analyzeFile("type_hints.py")
+
+        // type comments
+        val a = result.refs["a"]
+        assertNotNull(a)
+        assertEquals(result.assertResolvedType("int"), a.type)
```

Comment:
I'm not sure why I can't access assertResolvedType like in [PythonFrontendTest](https://github.com/Fraunhofer-AISEC/cpg/blob/4ee13d6156a98f877d09635c246442d7b458ffe5/cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt#L59C26-L59C44)

Reply:
assertResolvedType is an extension function of `MetadataProvider`. The reason behind this is that it needs access to the type manager (which can be accessed by the translation context). The only way to retrieve the context is through the metadata provider.

In order to use a syntax similar as in the other test is to use the Kotlin `with` function, which changes the specified object to be an implicit receiver within the block (black voodoo Kotlin magic).

```Kotlin
with(result) {
  assertResolvedType("int")
}
```

is equivalent to
```Kotlin
result.assertResolvedType("int")
```

Reply:
In order to align with the other tests, I would prefer the `with` syntax here.

Reply:
Done.